File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        int odd = 0;
        int even = 0;
        
        for (int num : num_list) {
            if (num % 2 == 0) {
                even *= 10;
                even += num;
            } else {
                odd *= 10;
                odd += num;
            }
        }
        
        return even + odd;
    }
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.kt
class Solution {
    fun solution(num_list: IntArray): Int {
        var odd = 0
        var even = 0
        
        num_list.forEach {
            if (it % 2 == 0) {
                even = even * 10 + it
            } else {
                odd = odd * 10 + it
            }
        }
        
        return odd + even
    }
}